deployment/deploy-az-dev-bootstrapper.ps1

```diff
@@ -5,37 +5,51 @@
+$l3LevelCoreInfra = ./deploy-core-infrastructure.ps1 -ParentConfig $l4LevelCoreInfra -ApplicationName ($ApplicationName + "L3") -VnetAddressPrefix "172.18.0.0/16" -SubnetAddressPrefix "172.18.0.0/18" -SetupArc $false -Location $Location
+$lowestLevelCoreInfra = ./deploy-core-infrastructure.ps1 -ParentConfig $l3LevelCoreInfra -ApplicationName ($ApplicationName + "L2") -VnetAddressPrefix "172.20.0.0/16" -SubnetAddressPrefix "172.20.0.0/18" -SetupArc $false -Location $Location
+
+# # 2. Deploy core platform in each layer (Dapr, Mosquitto and bridging).
+$l4CorePlatform = ./deploy-core-platform.ps1 -AksClusterName $l4LevelCoreInfra.AksClusterName -AksClusterResourceGroupName $l4LevelCoreInfra.AksClusterResourceGroupName -DeployDapr $true -MosquittoParentConfig $null
+$l3CorePlatform = ./deploy-core-platform.ps1 -AksClusterName $l3LevelCoreInfra.AksClusterName -AksClusterResourceGroupName $l3LevelCoreInfra.AksClusterResourceGroupName -MosquittoParentConfig $l4CorePlatform
+$l2CorePlatform = ./deploy-core-platform.ps1 -AksClusterName $lowestLevelCoreInfra.AksClusterName -AksClusterResourceGroupName $lowestLevelCoreInfra.AksClusterResourceGroupName -DeployDapr $true -MosquittoParentConfig $l3CorePlatform
 
-# Comment out below line if you are choosing the above 3 layer deployment instead.
-$lowestLevelCoreInfra = ./deploy-core-infrastructure.ps1 -ApplicationName ($ApplicationName + "L2") -VnetAddressPrefix "172.16.0.0/16" -SubnetAddressPrefix "172.16.0.0/18" -SetupArc $false
+# # 3. Deploy app resources in Azure, build images and deploy helm on level 4 and 2 (upper and lower).
+./deploy-dev-app.ps1 -ApplicationName $ApplicationName -AksClusterResourceGroupNameUpper $l4LevelCoreInfra.AksClusterResourceGroupName `
+    -AksClusterNameUpper $l4LevelCoreInfra.AksClusterName -AksServicePrincipalNameUpper ($ApplicationName + "L4") `
+    -AksClusterNameLower $lowestLevelCoreInfra.AksClusterName -AksClusterResourceGroupNameLower $lowestLevelCoreInfra.AksClusterResourceGroupName `
+    -AksServicePrincipalNameLower ($ApplicationName + "L2") -Location $Location
 
-# 2. Deploy core platform.
-./deploy-core-platform.ps1 -AksClusterName $lowestLevelCoreInfra.AksClusterName -AksClusterResourceGroupName $lowestLevelCoreInfra.AksClusterResourceGroupName
+# --- Deploying just a single layer: comment above block and uncomment below:
 
-# 3. Deploy app resources, build images and deploy helm.
-./deploy-dev-app.ps1 -ApplicationName $ApplicationName -AksClusterResourceGroupName $lowestLevelCoreInfra.AksClusterResourceGroupName -AksClusterName $lowestLevelCoreInfra.AksClusterName -AksServicePrincipalName ($ApplicationName + "L2")
+# $l4LevelCoreInfra = ./deploy-core-infrastructure.ps1 -ApplicationName ($ApplicationName + "L4") -VnetAddressPrefix "172.16.0.0/16" -SubnetAddressPrefix "172.16.0.0/18" -SetupArc $false -Location $Location

```

Comment:
Well spotted. This is because the same script is used once to execute per layer. On layers 2 and 4 we need Dapr, on layer 3 we don't so that is the reason for the parameter. If in the future we do deploy workloads with dapr on layer 3 then it's just a small change to enable it.

Reply:
...in deployment/deploy-core-platform.ps1 default value for $DeployDapr is $true. Does this mean Dapr is installed on L3 as well? Did you want to set default value to $false?

Reply:
You are so right! This needs to be set to $false by default. WIll update.